### PR TITLE
Additions to User Guide

### PR DESCRIPTION
**Pull request by blondegeek**
These are in progress changes to the User Guide that I'd like to get feedback on. I've added a message for the First Steps page, replaced a separate `irreps` discussion, and am working on an `interactions` page.

### CODE REVIEW

**Comment by mariogeiger** [comment]
Thanks for the RP!
Did you consider using the `.. jupyter-execute::` instruction into plain `.rst` files instead of `.ipynb` files?
Both execute the code in a notebook (under the hood)

See for example [here](https://docs.e3nn.org/en/latest/guide/change_of_basis.html) ([source](https://github.com/e3nn/e3nn/blob/main/docs/guide/change_of_basis.rst))

**Comment by blondegeek** [comment]
I was intending on prototyping in ipynb and then converting to rst but I
also understand if you would prefer previous commits to not contain those
files. I'm happy to work on this branch and then once finalized convert to
rst and commit to a separate branch that pulls directly from main.

**Comment by mariogeiger** [comment]
Ok perfect!
I think if I "Squash and merge" it will be fine, the `.ipynb` files will disappear

File: docs/guide/first_steps.ipynb [hunk 1]
```diff
@@ -4,71 +4,34 @@
...
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "mat = o3.rand_matrix()\n",
-    "irreps.D_from_matrix(mat)"
+    "Welcome to `e3nn`! We are happy you're here and want to help you hit the ground running in using `e3nn` in your projects.\n",
+    "\n",
+    "The goal of this first part of the User Guide is to give a conceptual framework for understanding `e3nn` and Euclidean neural networks. \n",
+    "\n",
+    "Euclidean neural networks are a tool to address the question -- suppose I want to train a neural network to do a task on 3D data, but I want it to understand that things \"mean the same thing\" if I change my coordinate system. This is especially important for tasks involving physical systems -- coordinate systems don't exist in the laws of physics, they are simply tools we use for or convenience so we can descibe a physical system e.g. to a computer.\n",
```
Comment (bdice) [inline docs/guide/first_steps.ipynb:20]:
```suggestion
    "Euclidean neural networks are a tool to address the question -- suppose I want to train a neural network to do a task on 3D data, but I want it to understand that things \"mean the same thing\" if I change my coordinate system. This is especially important for tasks involving physical systems -- coordinate systems don't exist in the laws of physics, they are simply tools we use for our convenience so we can descibe a physical system e.g. to a computer.\n",
```

File: docs/guide/first_steps.ipynb [hunk 2]
```diff
@@ -4,71 +4,34 @@
...
+    "\n",
+    "We have \"baked-in\" the assumption of Euclidean symmetry of 3D space into all neural network operations using representation theory -- the branch of mathematics that studies how do objects change under group action (in our case 3D rotations, translation, and inversion). Concepts like point groups, space groups, geometric tensors, and selection rules are consequences of representation theory.\n",
+    "\n",
+    "Because of this, something that is very different about `e3nn` from other neural networks is that `e3nn` operates on very strict data types defined by how they change under rotation and parity. We will go into more details about this in the section on irreducible representations or \"irreps\". \n",
+    "\n",
+    "All operations in `e3nn` handle tensor field data expressed in terms of irreps of the group of rotations and inversions, $O(3)$. This means that when you initialize modules, the module needs to know not only how many features to expect, but what types of features (e.g. 5 scalars and 7 vectors...).\n",
```
Comment (bdice) [inline docs/guide/first_steps.ipynb:26]:
```suggestion
    "All operations in `e3nn` handle tensor field data expressed in terms of irreps of the group of rotations and inversions, $O(3)$. This means that when you initialize modules, the module needs to know not only how many features to expect, but what types of features (e.g. 5 scalars and 7 vectors).\n",
```

File: docs/guide/first_steps.ipynb [hunk 3]
```diff
@@ -4,71 +4,34 @@
...
+    "\n",
+    "While two objects might have the same irreps, they can be interpretted to mean very different things. For example, a `1o` (vector-like) object can be interpretted as a regular 3D vector or as the coefficients spherical harmonics, signals on a sphere. Keep this freedom of interpretation in mind as we go through examples. For example, spherical harmonics are NOT irreps, but they do transform as irreps.\n",
```
Comment (bdice) [inline docs/guide/first_steps.ipynb:28]:
```suggestion
    "While two objects might have the same irreps, they can be interpreted to mean very different things. For example, a `1o` (vector-like) object can be interpreted as a regular 3D vector or as the coefficients of spherical harmonics, signals on a sphere. Keep this freedom of interpretation in mind as we go through examples. For example, spherical harmonics are NOT irreps, but they do transform in the same way as irreps.\n",
```

**Review by bdice — COMMENTED** [review]
A few small improvements I noticed while reading. I haven't looked at the irreps notebook (too big for GitHub to show a diff).

**Comment by blondegeek** [comment]
This is what I have been able to get done before taking off for vacation -- please feel free to do whatever you like with this pull request while I am gone. I know I'm leaving things with errors and I apologize!

**Comment by mariogeiger** [comment]
@blondegeek  reopen it if needed

**Pull request closed without merging**